### PR TITLE
gracefully handle change of questions which already has submissions

### PR DESCRIPTION
**Pull request by sheralim012**
- questionnaires having submissions of deleted questions can't be exported
- uses clean name for deleted questions in export

### CODE REVIEW

**Review by istride — CHANGES_REQUESTED** [review]
In general, I find these tests very hard to read and follow. There should be an effort to improve test code in the same way that regular application code should be improved too. One suggestion would be to write smaller, more focused tests, which should be simpler.

File: questionnaires/tests.py [hunk 1]
```diff
@@ -992,3 +992,65 @@ def test_page_permission(self):
         self.assertEqual(response.context['form_pages'][0].title, self.poll.title)
         self.assertEqual(response.context['form_pages'][1].title, self.survey.title)
         self.assertEqual(response.context['form_pages'][2].title, self.quiz.title)
+
+    def test_csv_export_with_modified_questions(self):
+        self.survey_question.delete()
+        skip_logic = json.dumps(
```
[the remark this answers is not in the document]
Comment (sheralim012) [reply questionnaires/tests.py]: done

File: questionnaires/tests.py [hunk 2]
```diff
@@ -992,3 +992,65 @@ def test_page_permission(self):
...
+        user_submission.submit_time = self.current_datetime
+        user_submission.save()
+        response = self.client.get(f'{self.url}?user_id={self.user_01.id}&export=csv')
+
+        byte_response = b''
+        for stream in response.streaming_content:
+            byte_response += stream
+        expected_response = \
+            f'ID,Name,Submission Date,Field,Value\r\n' \
+            f'{self.user_submission_01.id},{self.poll.title},{self.user_submission_01.submit_time},User,{self.user_01.username}\r\n' \
```
[the remark this answers is not in the document]
Comment (sheralim012) [reply questionnaires/tests.py]: submission id by database, questionnaire title and username by factory boy sequence generator, questionnaire URL by wagtail, and datetime from django timezone util all are dynamically generated

I think for questionnaire title, username, and datetime we can specify literal while creating and then use the same literal for assertions

but for submission id and questionnaire URL object should be used as these are out of our control

[the remark this answers is not in the document]
Comment (sheralim012) [reply questionnaires/tests.py]: done

**Review by istride — APPROVED** [review]

**Pull request merged**